crates/symbolicator/src/services/download/s3.rs

```diff
@@ -96,20 +99,38 @@ impl S3Downloader {
         } else {
             metric!(counter("source.s3.client.create") += 1);
 
-            let s3 = Arc::new(rusoto_s3::S3Client::new_with(
-                self.http_client.clone(),
-                rusoto_credential::StaticProvider::new_minimal(
-                    key.access_key.clone(),
-                    key.secret_key.clone(),
-                ),
-                key.region.clone(),
-            ));
+            let region = key.region.clone();
+            log::debug!(
+                "Using AWS credentials provider: {:?}",
+                key.aws_credentials_provider
+            );
+            let s3 = Arc::new(match key.aws_credentials_provider {
+                AwsCredentialsProvider::Container => {
+                    let provider = rusoto_credential::ContainerProvider::new();
+                    self.create_s3_client(provider, region)
+                }
+                _ => {
```

Comment:
An exhaustive match is preferably here, I don't think catch-all behaviour would be very future-proof.

Reply:
True; fixed.